src/events.cpp

```diff
@@ -245,6 +247,37 @@ ReturnValue Events::eventCreatureOnTargetCombat(Creature* creature, Creature* ta
 	return returnValue;
 }
 
+void Events::eventCreatureOnHear(Creature* creature, Creature* speaker, const std::string words, SpeakClasses type, Position pos)
+{
+	// Creature:onHear(speaker, words, type, pos)
+	if (info.creatureOnHear == -1) {
+		return;
+	}
+
+	if (!scriptInterface.reserveScriptEnv()) {
+		std::cout << "[Error - Events::eventCreatureOnHear] Call stack overflow" << std::endl;
+		return;
+	}
+
+	ScriptEnvironment* env = scriptInterface.getScriptEnv();
+	env->setScriptId(info.creatureOnHear, &scriptInterface);
+
+	lua_State* L = scriptInterface.getLuaState();
+	scriptInterface.pushFunction(info.creatureOnHear);
+
+	LuaScriptInterface::pushUserdata<Creature>(L, creature);
+	LuaScriptInterface::setCreatureMetatable(L, -1, creature);
+
+	LuaScriptInterface::pushUserdata<Creature>(L, speaker);
+	LuaScriptInterface::setCreatureMetatable(L, -1, speaker);
+
+	LuaScriptInterface::pushString(L, words);
+	lua_pushnumber(L, type);
+	LuaScriptInterface::pushPosition(L, pos);
```

Comment:
I don't see the purpose of the `pos` parameter. You can easily retrieve that information in the event by calling `speaker:getPosition()`.

Reply:
I know that you can do it that way aswell but sometimes it's not bad to have the extra param

Reply:
I agree, there is no need for it.
All you are doing is sending speaker:getPosition() but from C++ as creature->getPosition()
If the function is needed for a code we can do it via Lua.

Reply:
I'd have to agree with @TheSumm here - the position stuff is unnecessary.
The concept of this is fantastic though.